apps/teams/decorators.py

```diff
@@ -20,7 +21,10 @@ def login_and_team_required(view_func):
     @wraps(view_func)
     def _inner(request, *args, **kwargs):
         if not valid_auth_and_membership(request):
-            return HttpResponseRedirect("{}?next={}".format(reverse("account_login"), request.path))
+            next_url = request.get_full_path()
+            if not url_has_allowed_host_and_scheme(next_url, allowed_hosts={request.get_host()}):
```

Comment:
It feels like the `allowed_hosts` param should come from `settings` instead. What am I missing?

Reply:
these changes were suggested by CodeQL. We do similar things in other places.